addons/point_of_sale/models/pos_session.py

```diff
@@ -1550,6 +1553,419 @@ def try_cash_in_out(self, _type, amount, reason, extras):
             message_content.append(f'- Reason: {reason}')
         self.message_post(body='<br/>\n'.join(message_content))
 
+    def get_onboarding_data(self):
+        return {
+            "categories": self._load_model('pos.category'),
+            "products": self._load_model('product.product'),
+        }
+
+    def _load_model(self, model):
+        model_name = model.replace('.', '_')
+        if hasattr(self, '_get_pos_ui_%s' % model_name) and hasattr(self, '_loader_params_%s' % model_name):
+            return getattr(self, '_get_pos_ui_%s' % model_name)(getattr(self, '_loader_params_%s' % model_name)())
```

Comment:
Seems unnecessary to duplicate the lookup (and formatting), could be e.g.
```suggestion
        loader = getattr(self, '_get_pos_ui_%s' % model_name, None)
        params = getattr(self, '_loader_params_%s' % model_name, None)
        if loader and params:
            return loader(params())
```